scene/gui/tab_bar.cpp

```diff
@@ -827,78 +828,52 @@ void TabBar::_update_cache() {
 	int limit_minus_buttons = limit - incr->get_width() - decr->get_width();
 
 	int w = 0;
-	int mw = 0;
-	int size_fixed = 0;
-	int count_resize = 0;
 
-	for (int i = 0; i < tabs.size(); i++) {
-		tabs.write[i].size_text = Math::ceil(tabs[i].text_buf->get_size().x);
-		tabs.write[i].text_buf->set_width(-1);
+	max_drawn_tab = tabs.size() - 1;
 
-		tabs.write[i].ofs_cache = 0;
+	for (int i = 0; i < tabs.size(); i++) {
+		tabs.write[i].size_text = tabs[i].size_text_og;
+		tabs.write[i].text_buf->set_width(tabs[i].size_text_og);
```

Comment:
Not sure if it's necessary to save `size_text_og`, setting it to `-1` (unlimited) and getting size will do the same. Also, checking for `max_width`, probably can be done here.

```c++
int text_width = (max_width > 0) ? max_width - ICON_AND_STYLE_WIDTH : -1;
tabs.write[i].text_buf->set_width(text_width);
tabs.write[i].size_text = Math::ceil(tabs[i].text_buf->get_size().x);
```

Reply:
Thanks for the tip. `size_text_og` has been removed.